### ISSUE TITLE
plane fitting

### ISSUE THREAD

**Issue by Sun-god-pineapple** [zjjMaiMai/TinyHITNet#14]
Hi, Thank you so much for your wonderful contribution!
I'm having some trouble understanding "plane_fitting_sf.py", especially this function down here
![TR@F)F R %ZAW(UYN$C5%6](https://user-images.githubusercontent.com/87853024/165216826-9ada3efe-8fbc-4ff1-99ed-1b93e5e1c80d.png)
Could you please explain this function for me and what part of the paper it corresponds to at your convenience?
Thank you for your time!

**Comment by zjjMaiMai** [comment]
we compute dx/dy by RANSAC plane fitting offline and implemented on cuda.  
which described in paper **4.2. Propagation Loss** :  
> We use the ground truth disparity d^{gt} and ground truth disparity gradients d_{x}^{gt} and d_{y}^{gt}, which we compute by robustly fitting a plane to d^{gt} in a 9x9 window centered at the pixel.

**Comment by Sun-god-pineapple** [comment]
> we compute dx/dy by RANSAC plane fitting offline and implemented on cuda. which described in paper **4.2. Propagation Loss** :
> 
> > We use the ground truth disparity d^{gt} and ground truth disparity gradients d_{x}^{gt} and d_{y}^{gt}, which we compute by robustly fitting a plane to d^{gt} in a 9x9 window centered at the pixel.

Thank you so much for your reply!
I still have some questions down here.
1. What is the meaning of parameter "sigma"?
2. I found that you choose the defferent "iter" and "sigma" for SceneFlow Dataset and KITTI Dataset, could you please explain how you chose them ?
3. I am trying to use your codes on another dataset, but some ground truth disparities in that dataset are very sparse. For example,  some disparity may only have 10% of the pixels that have ground truth, while the remaing pixels have a value of 0. In that case, how can I set the parameters of the plane fitting so that the model can train good results.
Looking forward to your reply at your convenience!

**Comment by zjjMaiMai** [comment]
> 1. What is the meaning of parameter "sigma"?

sigma is a hyperparameter in [RANSAC](https://en.wikipedia.org/wiki/Random_sample_consensus). that samples which errors less than sigma as "inliers".

> 2. I found that you choose the defferent "iter" and "sigma" for SceneFlow Dataset and KITTI Dataset, could you please explain how you chose them ?
> 3. I am trying to use your codes on another dataset, but some ground truth disparities in that dataset are very sparse. For example, some disparity may only have 10% of the pixels that have ground truth, while the remaing pixels have a value of 0. In that case, how can I set the parameters of the plane fitting so that the model can train good results.
Looking forward to your reply at your convenience!

we tune these hyperparameters by observe quality of the [visualized dx/dy](https://github.com/zjjMaiMai/TinyHITNet/blob/master/colormap.py#L25-L31). you can try this step by your self and adjust hyperparameters. i think too sparse ground truth can not get a good dx/dy estimation.

**Comment by Sun-god-pineapple** [comment]
Thank you so much for your kind reply, It is really helpful!
Wishing you a beautiful day!

**Comment by zhangjy328** [comment]
> we tune these hyperparameters by observe quality of the [visualized dx/dy](https://github.com/zjjMaiMai/TinyHITNet/blob/master/colormap.py#L25-L31).

I also found that you choose the defferent "iter" and "sigma" for SceneFlow Dataset and KITTI Dataset:
`x = HitnetModule.plane_fitting(x, 1024, 1, 9, 1e-3, 1e5)`
`x = HitnetModule.plane_fitting(x, 256, 0.1, 9, 0, 1e5)`

You said that these hyperparameters can be tuned by observing the quality of the visualized dx/dy. I'd like to ask how to evaluate whether it's good or bad. What kind of image is considered to be better? At the same time, the parameters of these two datasets vary greatly. What basis do I have to determine a rough adjustment range for my own dataset?For example, is the following picture good or bad?

![Image](https://github.com/user-attachments/assets/e3526784-2fb2-4364-9c08-7fab6c48ec97)
